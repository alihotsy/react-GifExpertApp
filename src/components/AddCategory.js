import React, {useState} from 'react'
import PropTypes from 'prop-types';

const useMyInput = (setCategories) =>{
    const [inputValue,setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleInput = (e) =>{
        if(e.key === "Enter"){
            if(e.target.className === "input1"){
                if(inputValue.trim().length > 2){
                    setCategories(cats => [inputValue,...cats]);
                    setInputValue('');
                }
            } else{
                setCategories(cats => cats.filter(n => n !== inputValue))
                setInputValue('');
            }
           
        }
    }
    const handleSubmit = (e) =>{
        e.preventDefault();
        
    }
    return {
        handleInputChange,
        handleSubmit,
        inputValue,
        handleInput
    }
}
export const AddCategory = ({setCategories}) => {
    const {handleSubmit} = useMyInput(setCategories)
    const input1 = useMyInput(setCategories);
    const input2 = useMyInput(setCategories);
    return (
        <form onSubmit = {handleSubmit}> 
            <p>{input1.inputValue}</p>
            <label>Ingresa una serie:</label>
            <input
             className = "input1"
             type="text"
             value = {input1.inputValue}
             onChange = {input1.handleInputChange}
             onKeyPress = {input1.handleInput}
            />
            <label>Borra una serie de la lista:</label>
            <input
             type="text2"
             value = {input2.inputValue}
             onChange = {input2.handleInputChange}
             onKeyPress = {input2.handleInput}
            />
            <button className="boton">Borrar</button>
        </form>    
        
        
    )
}
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}