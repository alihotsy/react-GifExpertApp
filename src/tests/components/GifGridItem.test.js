import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
describe('Pruebas en <GifGridItem.js/>', () => {
        const title = 'Un título';
        const url =  'https://localhost/algo.jpg';
        const wrapper = shallow(<GifGridItem title={title} url={url}/>)
    test('Debe mostrar el componente <GifGridItem.js/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    })
   test('Debe de tener un párrafo con el title', () => {
       const p = wrapper.find('p').text();
       expect(p).toBe(title)
   })
   test('Debe de tener una imagen igual al url y alt de los props', () => {
       const img = wrapper.find('img');
       expect(img.prop('src')).toBe(url);
       expect(img.prop('alt')).toBe(title);
   })
   test('El div debe tener un animate__fadeIn', () => {
       const div = wrapper.find('div');
       const className = div.prop('className');
       expect(className.includes('animate__fadeIn')).toBe(true);
   })
   
})
