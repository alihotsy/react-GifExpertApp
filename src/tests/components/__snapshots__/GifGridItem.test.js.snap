// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem.js/> Debe mostrar el componente <GifGridItem.js/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    className="card-img"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
